### PR TITLE
Make changes to support go mobile.

### PR DESCRIPTION
**Pull request by jigar-f**

### CODE REVIEW

File: minisql/values.go [hunk 1]
```diff
@@ -75,13 +66,8 @@ type Values interface {
...
 }
 
-func NewValues(vals []interface{}) Values {
-	values := make(valueArray, len(vals))
-	for i, v := range vals {
-		values[i] = NewValue(v)
-	}
-	return &valueArrayWrapper{values}
-}
+// Removed generic interface{} type for value array creation
```
Comment (oxtoacart) [inline minisql/values.go:69]: This comment doesn't need to be in the code and will just be confusing to someone looking at the code, please move comment into PR instead.

**Comment by oxtoacart** [comment]
This PR breaks the tests (you can run the tests with `make test`).

```
--- FAIL: TestTransactions (0.01s)
panic: unsupported type provided: *[]uint8 [recovered]
	panic: unsupported type provided: *[]uint8

goroutine 6 [running]:
testing.tRunner.func1.2({0x1005377a0, 0xc0000778e0})
	/usr/local/go1.19.5/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
	/usr/local/go1.19.5/src/testing/testing.go:1399 +0x39f
panic({0x1005377a0, 0xc0000778e0})
	/usr/local/go1.19.5/src/runtime/panic.go:884 +0x212
github.com/getlantern/pathdb/minisql.convertToValueSlice({0xc0000778d0, 0x1, 0x10000f867?})
	/Users/ox/gocode/src/github.com/getlantern/pathdb/minisql/api.go:77 +0x405
github.com/getlantern/pathdb/minisql.(*scannableRows).Scan(0xc000077880, {0xc0000778d0?, 0x2a?, 0xc0000bdc60?})
	/Users/ox/gocode/src/github.com/getlantern/pathdb/minisql/api.go:62 +0x30
github.com/getlantern/pathdb.(*queryable).get(0xc000025040, {0x1005abb89?, 0x10000d196?})
	/Users/ox/gocode/src/github.com/getlantern/pathdb/db.go:227 +0x244
github.com/getlantern/pathdb.GetOrPut[...]({0x10064f660, 0xc000025040}, {0x1005abb89, 0x4}, {0x1005b6be6, 0x1d}, {0x0, 0x0})
	/Users/ox/gocode/src/github.com/getlantern/pathdb/db_interface.go:72 +0x6f
github.com/getlantern/pathdb.TestTransactions.func1.1({0x10064f660, 0xc000025040})
	/Users/ox/gocode/src/github.com/getlantern/pathdb/db_test.go:28 +0x205
github.com/getlantern/pathdb.Mutate({0x10064f610?, 0xc000099ba0?}, 0xc0000bde88)
	/Users/ox/gocode/src/github.com/getlantern/pathdb/db_interface.go:26 +0x49
github.com/getlantern/pathdb.TestTransactions.func1({0x10064f610?, 0xc000099ba0})
	/Users/ox/gocode/src/github.com/getlantern/pathdb/db_test.go:20 +0x65
github.com/getlantern/pathdb.withDB(0x30a9b3e000062758?, 0xc0000bdf50)
	/Users/ox/gocode/src/github.com/getlantern/pathdb/db_test.go:478 +0x162
github.com/getlantern/pathdb.TestTransactions(0x0?)
	/Users/ox/gocode/src/github.com/getlantern/pathdb/db_test.go:18 +0x35
testing.tRunner(0xc0000f4820, 0x1005e0620)
	/usr/local/go1.19.5/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	/usr/local/go1.19.5/src/testing/testing.go:1493 +0x35f
FAIL	github.com/getlantern/pathdb	0.742s
?   	github.com/getlantern/pathdb/minisql	[no test files]
FAIL
```

**Comment by jigar-f** [comment]
Thanks @oxtoacart, For the review, But the Last changes we did, were not supported in go mobile, We were using potiners in values, Now we are using this,

```
type Value struct {
	Type   int
	String string
	Int    int
	Bytes  string
}
```

So looking at this do you think there is room for improvement, or should I just make changes on the test case and merged this to the main?

**Comment by oxtoacart** [comment]
We can't use pointers with iOS? Also we can't use []byte? Converting bytes to strings is pretty inefficient.

**Comment by jigar-f** [comment]
This is not needed anymore.

**Pull request closed without merging**